jetty-jspc-maven-plugin/src/main/java/org/eclipse/jetty/jspc/plugin/JspcMojo.java

```diff
@@ -241,12 +242,6 @@ protected TldScanner newTldScanner(JspCServletContext context, boolean namespace
     @Parameter
     private String targetVersion;
 
-    /**
-     * The JspC instance being used to compile the jsps.
```

Comment:
I think we should leave this as a parameter  - that way, users can configure it from the pom if they want.

Reply:
agree and not agree :) 
Because even if a user provide his own from a pom we will override some values.
Look at the code https://github.com/eclipse/jetty.project/blob/7555d0375aac6e0d9d51b0785d5259d80af3905d/jetty-jspc-maven-plugin/src/main/java/org/eclipse/jetty/jspc/plugin/JspcMojo.java#L336.
it's not very consistent and can be confusing for user who provide their own jspc as values will be override by some default values.

Reply:
Sure, however, the values that we are setting explicitly are able to be configured by params we've added to the plugin to integrate it with the jetty/unassembled webapp environment. There are dozens of other setters that exist on the JspC class that the user should be able to have access to that we don't want to have to add params to our plugin for - moreover, it's a bit of future-proofing: the JspC class may well change to add more setters and users would be able to access them without waiting for us to add params to the plugin.

Reply:
fair enough but as a user I find it very confusing.
Using this:
```
<configuration>
  <jspc>
    <webXmlInclude>foo/bar/blabla.xml</webXmlInclude>
    few other attributes
  </jspc>
</configuration>
```
At the end the value used for `jspc.webXmlInclude` will be `${basedir}/target/webfrag.xml` because we always override with mojo field.
We should fixed that (well noboby complained so it's probably not an issue ;) ).
But what about `merging` the configuration? Pseudo code:
```
if jspc!=null && jsp.webXmlInclude1=null {
  // do not use value from mojo field
}
```
WDYT?

Reply:
not related to this PR I opened another issue https://github.com/eclipse/jetty.project/issues/6235

Reply:
@janbartel change reverted